krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -0,0 +1,240 @@
+/*
+ Copyright 2023 The Nephio Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// errors
+	errKubeObjectNotInitialized = "KubeObject not initialized"
+)
+
+var (
+	attachmentType      = []string{"spec", "attachmentType"}
+	cniType             = []string{"spec", "cniType"}
+	networkInstanceName = []string{"spec", "networkInstance", "name"}
+)
+
+type Interface interface {
+	// GetKubeObject returns the present kubeObject
+	GetKubeObject() *fn.KubeObject
+	// GetGoStruct returns a go struct representing the present KRM resource
+	GetGoStruct() (*nephioreqv1alpha1.Interface, error)
+	// GetAttachmentType returns the attachmentType from the spec
+	// if an error occurs or the attribute is not present an empty string is returned
+	GetAttachmentType() string
+	// GetCNIType returns the cniType from the spec
+	// if an error occurs or the attribute is not present an empty string is returned
+	GetCNIType() string
+	// GetNetworkInstanceName returns the name of the networkInstance from the spec
+	// if an error occurs or the attribute is not present an empty string is returned
+	GetNetworkInstanceName() string
+	// SetAttachmentType sets the attachmentType in the spec
+	SetAttachmentType(s string) error
+	// SetCNIType sets the cniType in the spec
+	SetCNIType(s string) error
+	// SetNetworkInstanceName sets the name of the networkInstance in the spec
+	SetNetworkInstanceName(s string) error
+	// SetSpec sets the spec attributes in the kubeobject according the go struct
+	SetSpec(*nephioreqv1alpha1.InterfaceSpec) error
+	// DeleteAttachmentType deletes the attachmentType from the spec
+	DeleteAttachmentType() error
+	// DeleteAttachmentType deletes the attachmentType from the spec
+	DeleteCNIType() error
+}
+
+// NewFromYAML creates a new parser interface
+// It expects a raw byte slice as input representing the serialized yaml file
+func NewFromYAML(b []byte) (Interface, error) {
+	o, err := fn.ParseKubeObject(b)
```

Comment:
question: Looking at the fn.ParseKubeObject(), it seems like `o` will never be `nil`. Is this assumption valid?